### PR TITLE
Make DMX Rx work on Teensy 3.2

### PR DESCRIPTION
**Pull request by peternewman**
Only tested on Serial1 so far

### CODE REVIEW

File: TeensyDmx.cpp [hunk 1]
```diff
@@ -1040,17 +1040,36 @@ void TeensyDmx::startReceive()
...
+        // Fire UART2 receive interrupt immediately after each byte received
+        UART2_RWFIFO = 1;
+
+        // Enable UART2 interrupt on frame error and enable IRQ
+        UART2_C3 |= UART_C3_FEIE;
+        NVIC_ENABLE_IRQ(IRQ_UART2_STATUS);
+
         attachInterruptVector(IRQ_UART2_STATUS, UART2RxStatus);
     }
+    // TODO: Does this need to handle Serial4-6 here too?
```
Comment (peternewman) [inline TeensyDmx.cpp:1072]: Does it need doing, or do they only support IRQ_UART0_ERROR method?

**Comment by peternewman** [comment]
Tx on Serial1 still works too!

This was guesswork/debugging, so I may have added some stuff that's unnecessary, but I don't think so.

**Comment by peternewman** [comment]
While DMX in currently works, RDM doesn't...

**Comment by peternewman** [comment]
If you're happy with this, you may prefer to just merge #13 which includes this and the RDM responder fixes and shouldn't need a squash.

**Comment by peternewman** [comment]
Bump @chrisstaite . Have you got any time to review this and/or #13 ? I've got a few more improvements regards RDM Rx packet validation to follow too.

**Comment by chrisstaite** [comment]
This pull request looks great, but has conflicts with the head so I can't simply re-base and merge.  I'm happy to squash and merge if you like, or you can re-base on the current master and I'll merge as-is.  I'll leave it to you.  Thanks for your work with this, I've been far too busy to crack out my Teensy's on the test bench this summer!

File: TeensyDmx.cpp (hunk 1 above)
Comment (peternewman) [reply TeensyDmx.cpp:1072]: Did you see this query @chrisstaite ?

**Comment by peternewman** [comment]
I think I've fixed the issue now, but I'm not certain. If it's not working, the best bet may be to just squash!

**Comment by chrisstaite** [comment]
Still getting "This branch cannot be rebased due to conflicts".

Happy to squash, but I'd like to credit the commits to your authorship.
Do you need any help with your git use, or is that not the problem?  Happy to help if you need it.

File: TeensyDmx.cpp (hunk 1 above)
Comment (chrisstaite) [reply TeensyDmx.cpp:1072]: I saw it, but wasn't sure of the answer myself, therefore happy to leave it as a comment for someone to take on (possibly me) in the future.  It's actually chip dependant, and when I started it required a fair amount of reading of the datasheet to figure out which was which and for which chip...

Comment (peternewman) [reply TeensyDmx.cpp:1072]: I've got access to a Teensy 3.5 too now, so when I get a chance, I'll try and test that on all it's serials.

**Comment by peternewman** [comment]
I can't see anything the other way now on GitHub:
https://github.com/peternewman/TeensyDmx/compare/master...chrisstaite:master?expand=1

TBH this is my first time using rebase, so it may well be an issue my end, I normally just do "git pull upstream master", or even use the "Out of Date" buttons within GitHub (not sure if that's a permissions thing, or just a setting on the repo).

I think GitHub shows the authorship of a squash anyway. Alternatively you could try this cherry-pick:
https://github.com/chrisstaite/TeensyDmx/pull/14

**Comment by peternewman** [comment]
Actually, the bottom of the PR says the same thing, "This branch has no conflicts with the base branch", so I'm rather confused if you're seeing conflicts.

File: TeensyDmx.cpp (hunk 1 above)
Comment (chrisstaite) [reply TeensyDmx.cpp:1072]: Sounds good, happy to add this support later though.

**Comment by chrisstaite** [comment]
Pulled as #14

**Pull request closed without merging**